### DISCUSSION TITLE
Unable to retrieve data on `BelongsTo` inside a `BelongsToMany` fields in edit mode

### DISCUSSION THREAD

**Discussion by kichetof** [laravel/nova-issues#5079] [Q&A]
- Laravel Version: 9.40.1
- Nova Version: 4.19.2
- PHP Version: 8.1.11
- Database Driver & Version: Sail
- Operating System and Version: Docker

### Description:
When you have a `BelongsTo` field in a `BelongsToMany` fields, data aren't retrieved in attach mode: 404 error.
Everything else works perfectly (index data, inline creation)

That's the url with 404 error:
http://localhost/nova-api/collaborators/associatable/job?first=false&search=&withTrashed=false&resourceId=1312&viaResource=collaborators&viaResourceId=1312&viaRelationship=units&component=belongsto.belongs-to-field.job&dependsOn=eyJqb2IiOiIifQ%3D%3D&editing=true&editMode=update

Workaround is to retrieve data and add a `Select` field with `onlyOnForms` but we loose the ability to create inline...

### Detailed steps to reproduce the issue on a fresh Nova installation:

```php
// collaborator resource field
BelongsToMany::make('Unité', 'units', Unit::class)
                ->fields(new CollaboratorUnitFields)
                ->allowDuplicateRelations(),

// CollaboratorUnitFields
BelongsTo::make('Poste', 'job', Job::class)
                ->showCreateRelationButton()
                ->withoutTrashed(),
```

**Comment by crynobone** [comment]
Eloquent doesn't provide any support to query pivot relationship with relationship. You can do it manually but since Nova depends heavily on Eloquent we unable to provide this without Eloquent first allow nested query.

**Reply by kichetof** [reply]
Thanks for your feedback! Maybe this information should be indicated in the doc as everything works well except form view?
